Description: Mount drives in /media, not /run/media/, to stay FHS compatible. As on Debian and Ubuntu /media is not currently a tmpfs, we need to put the "mounted-fs" file to a persistent path as well.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=51709
Bug-Debian: http://bugs.debian.org/680403
Bug-Ubuntu: https://launchpad.net/bugs/1020759
diff -Naur udisks-2.0.92/src/udiskslinuxfilesystem.c udisks-2.0.92.tpg/src/udiskslinuxfilesystem.c
--- udisks-2.0.92/src/udiskslinuxfilesystem.c	2013-02-17 22:46:46.000000000 +0000
+++ udisks-2.0.92.tpg/src/udiskslinuxfilesystem.c	2013-02-20 19:19:29.382329401 +0000
@@ -877,26 +877,26 @@
     }
 
   /* If we know the user-name and it doesn't have any '/' character in
-   * it, mount in /run/media/$USER
+   * it, mount in /media/$USER
    */
   if (!fs_shared && (user_name != NULL && strstr (user_name, "/") == NULL))
     {
-      mount_dir = g_strdup_printf ("/run/media/%s", user_name);
+      mount_dir = g_strdup_printf ("/media/%s", user_name);
       if (!g_file_test (mount_dir, G_FILE_TEST_EXISTS))
         {
-          /* First ensure that /run/media exists */
-          if (!g_file_test ("/run/media", G_FILE_TEST_EXISTS))
+          /* First ensure that /media exists */
+          if (!g_file_test ("/media", G_FILE_TEST_EXISTS))
             {
-              if (g_mkdir ("/run/media", 0755) != 0)
+              if (g_mkdir ("/media", 0755) != 0)
                 {
                   g_set_error (error,
                                UDISKS_ERROR,
                                UDISKS_ERROR_FAILED,
-                               "Error creating directory /run/media: %m");
+                               "Error creating directory /media: %m");
                   goto out;
                 }
             }
-          /* Then create the per-user /run/media/$USER */
+          /* Then create the per-user /media/$USER */
           if (g_mkdir (mount_dir, 0700) != 0)
             {
               g_set_error (error,
